### ISSUE TITLE
[Impeller] Implement Wide Gamut colors for Vulkan

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#127852]
As a result of https://github.com/flutter/flutter/issues/55092, we now have wide gamut image / gradient support on iOS.  As Vulkan Impeller support matures we should add matching support in the vulkan implementation.

**Comment by deckerst** [comment]
@gaaclarke @chinmaygarde would this wide gamut support also allow Flutter apps to properly display Ultra HDR images?

**Comment by gaaclarke** [comment]
> @gaaclarke @chinmaygarde would this wide gamut support also allow Flutter apps to properly display Ultra HDR images?

No, I don't believe so.  Ultra HDR is a different file format that would have to be supported, which is unrelated to the rendering code.  If that is supported I think wide gamut rendering may make rendering Ultra HDR images more correct since we have more colors to work with, and thus more levels of contrast.  It all depends on how smart the compatibility with the format the decoder is.  For example, it seems like an Ultra HDR image could be in sRGB, but the gain map in the file will boost the contrast.  If the decoder just throws away the gain map, it will be no difference.  If it maps it to the closest color in the display p3 gamut, that's a different story.

I've never rendered HDR content, but it seems like wide gamut could help, but doesn't mean we'll get support for Ultra HDR or renderer it 100% as it was intended without special consideration in the renderer.

**Comment by deckerst** [comment]
Thanks for the quick response.

I'm aware of the format and how it embeds a gain map, but I see from Android native apps that they don't need to extract or apply it themselves. On Android 14, one only needs [to decode the file as usual via BitmapFactory](https://github.com/android/platform-samples/blob/main/samples/graphics/ultrahdr/src/main/java/com/example/platform/graphics/ultrahdr/display/DisplayUltraHDRScreen.kt#L92), and [to set the display color mode](https://github.com/android/platform-samples/blob/main/samples/graphics/ultrahdr/src/main/java/com/example/platform/graphics/ultrahdr/display/DisplayUltraHDRScreen.kt#L150).

When I try to do so in a Flutter app (setting the color mode & decoding the image on the platform side first), the display does switch to the HDR color mode, but the image is still not displayed as it should. I wasn't sure whether it was a limitation of Flutter rendering or an issue on my side.

**Comment by jonahwilliams** [comment]
We are not currently working on this
